pkg/server/options/authorization.go

```diff
@@ -0,0 +1,112 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package options
+
+import (
+	"github.com/spf13/pflag"
+
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+	"k8s.io/apiserver/pkg/authorization/authorizerfactory"
+	"k8s.io/apiserver/pkg/authorization/path"
+	"k8s.io/apiserver/pkg/authorization/union"
+	genericapiserver "k8s.io/apiserver/pkg/server"
+	coreexternalversions "k8s.io/client-go/informers"
+
+	"github.com/kcp-dev/kcp/pkg/authorization"
+	"github.com/kcp-dev/kcp/pkg/client/listers/tenancy/v1alpha1"
+)
+
+type Authorization struct {
+	// AlwaysAllowPaths are HTTP paths which are excluded from authorization. They can be plain
+	// paths or end in * in which case prefix-match is applied. A leading / is optional.
+	AlwaysAllowPaths []string
+
+	// AlwaysAllowGroups are groups which are allowed to take any actions.  In kube, this is system:masters.
+	AlwaysAllowGroups []string
+}
+
+func NewAuthorization() *Authorization {
+	return &Authorization{
+		// This allows the kubelet to always get health and readiness without causing an authorization check.
+		// This field can be cleared by callers if they don't want this behavior.
+		AlwaysAllowPaths:  []string{"/healthz", "/readyz", "/livez"},
+		AlwaysAllowGroups: []string{"system:masters"},
+	}
+}
+
+// WithAlwaysAllowGroups appends the list of paths to AlwaysAllowGroups
+func (s *Authorization) WithAlwaysAllowGroups(groups ...string) *Authorization {
+	s.AlwaysAllowGroups = append(s.AlwaysAllowGroups, groups...)
+	return s
+}
+
+// WithAlwaysAllowPaths appends the list of paths to AlwaysAllowPaths
+func (s *Authorization) WithAlwaysAllowPaths(paths ...string) *Authorization {
+	s.AlwaysAllowPaths = append(s.AlwaysAllowPaths, paths...)
+	return s
+}
+
+func (s *Authorization) Validate() []error {
+	if s == nil {
+		return nil
+	}
+
+	allErrors := []error{}
+
+	return allErrors
+}
+
+func (s *Authorization) AddFlags(fs *pflag.FlagSet) {
+	if s == nil {
+		return
+	}
+
+	fs.StringSliceVar(&s.AlwaysAllowPaths, "authorization-always-allow-paths", s.AlwaysAllowPaths,
+		"A list of HTTP paths to skip during authorization, i.e. these are authorized without "+
+			"contacting the 'core' kubernetes server.")
+}
+
+func (s *Authorization) ApplyTo(config *genericapiserver.Config, informer coreexternalversions.SharedInformerFactory, workspaceLister v1alpha1.WorkspaceLister) error {
+	var authorizers []authorizer.Authorizer
+
+	// group authorizer
+	if len(s.AlwaysAllowGroups) > 0 {
+		authorizers = append(authorizers, authorizerfactory.NewPrivilegedGroups(s.AlwaysAllowGroups...))
+	}
+
+	// path authorizer
+	if len(s.AlwaysAllowPaths) > 0 {
```

Comment:
@s-urbaniak why was this not needed before? Without `/readyz` only works authenticated.

Reply:
it seems it was not wired before:
```
% curl -k https://127.0.0.1:6443/readyz
Internal Server Error: "/readyz": expected logical cluster name to be in org_name format, got admin
```